### PR TITLE
Consider error from `WorkerDelegate` for ErrorCodes

### PR DESCRIPTION
**Pull request by acumino**
**How to categorize this PR?**

/area ops-productivity
/kind enhancement

**What this PR does / why we need it**:
Consider errors from `WorkerDelegate` for error codes so the errors can be classified properly.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```other operator
NONE
```

### CODE REVIEW

**Review by Kostov6 — COMMENTED** [review]
Thanks for the PR! Just a couple of questions from my side:
1. What does `actuator context` refer to? 
2. Is this the only function in the file that may return ErrorCodes? If yes how does one know if a given function may return ErrorCodes?

**Comment by acumino** [comment]
> 1. What does `actuator context` refer to?

It was there before, so I kept it, but it mostly meant, all the things needed for the worker actuator, like azure client.

> 2. Is this the only function in the file that may return ErrorCodes? If yes how does one know if a given function may return ErrorCodes?

Based on what the function is doing, if it is dealing with the provider-specific clients then yes, the error returned should be taken in consideration for error codes.

File: extensions/pkg/controller/worker/genericactuator/actuator_delete.go [hunk 1]
```diff
@@ -35,7 +35,11 @@ func (a *genericActuator) Delete(ctx context.Context, log logr.Logger, worker *e
 
 	workerDelegate, err := a.delegateFactory.WorkerDelegate(ctx, worker, cluster)
 	if err != nil {
-		return fmt.Errorf("could not instantiate actuator context: %w", err)
+		newError := fmt.Errorf("could not instantiate actuator context: %w", err)
```
Comment (Kostov6) [inline extensions/pkg/controller/worker/genericactuator/actuator_delete.go]:
```suggestion
		newError := fmt.Errorf("could not instantiate actuator: %w", err)
```
When seeing `context` in error message I associate it with `context.Context` and `actuator context` reads to me as the function's fail is related to it's context variable. If `context` does not provide any benefit in the error message let's drop it

**Review by Kostov6 — COMMENTED** [review]
/lgtm

**Pull request merged**